From cdfb0b0656b6fd371cf73333af54530e8de556e1 Mon Sep 17 00:00:00 2001
From: Colin Ian King <colin.i.king@gmail.com>
Date: Fri, 3 May 2024 10:46:01 +0100
Subject: [PATCH] core-asm-arm.h: declare stress_asm_arm_yield when
 HAVE_ASM_ARM_YIELD is defined

Don't declare inlined yield helper if yield is not available

Signed-off-by: Colin Ian King <colin.i.king@gmail.com>
---
 core-asm-arm.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/core-asm-arm.h b/core-asm-arm.h
index c275dc8b9bda..152e771811b8 100644
--- a/core-asm-arm.h
+++ b/core-asm-arm.h
@@ -24,10 +24,12 @@
 
 #if defined(STRESS_ARCH_ARM)
 
+#if defined(HAVE_ASM_ARM_YIELD)
 static inline void ALWAYS_INLINE stress_asm_arm_yield(void)
 {
 	__asm__ __volatile__("yield;\n");
 }
+#endif
 
 /* #if defined(STRESS_ARCH_ARM) */
 #endif
-- 
2.45.0

